### PR TITLE
refactor: Extract cni manager as the public part

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Starnop <starnop@163.com>

### Ⅰ. Describe what this PR did
For now，The organization of the code about CNI Manager is unreasonable，we should reorganize the file structure: extract cni manager as the public part.

### Ⅱ. Does this pull request fix one issue?

None.

### CODE REVIEW

File: cri/ocicni/cni_manager.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+// are launched.
+func (c *CniManager) SetUpPodNetwork(podNetwork *ocicni.PodNetwork) error {
+	_, err := c.plugin.SetUpPod(*podNetwork)
+
+	defer func() {
+		if err != nil {
+			// Teardown network if an error returned.
+			err := c.plugin.TearDownPod(*podNetwork)
+			if err != nil {
+				logrus.Errorf("failed to detroy network for sandbox %q: %v", podNetwork.ID, err)
```
Comment (allencloud) [inline cri/ocicni/cni_manager.go]: s/detroy/destroy

File: cri/ocicni/types.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package ocicni
```
Comment (allencloud) [inline cri/ocicni/types.go]: I do not think defining an interface should use a filename of `types.go`.
Instead, maybe cnimanager.go is a better one. WDYT?

**Comment by allencloud** [comment]
Sounds good to me, but some nits still need update. @Starnop

File: cri/ocicni/types.go (hunk 2 above)
Comment (Starnop) [reply cri/ocicni/types.go]: `cnimanager.go` doesn't make sense, there is a file named `cni_manager.go`. So maybe `apis.go` or something else?

Comment (allencloud) [reply cri/ocicni/types.go]: @YaoZengzeng 
Any advice?

Comment (YaoZengzeng) [reply cri/ocicni/types.go]: I think `types.go` is fine, we could expand it if we need more types 😄

File: cri/ocicni/cni_manager.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
+package ocicni
+
+import (
+	"fmt"
+	"github.com/alibaba/pouch/cri/config"
```
[the remark this answers is not in the document]
Comment (Starnop) [reply cri/ocicni/cni_manager.go]: Done.

File: cri/v1alpha1/cri_network.go [hunk 4]
```diff
@@ -1,130 +1,12 @@
...
-	return ip, nil
-}
-
-// Status returns error if the network plugin is in error state.
-func (c *CniManager) Status() error {
-	return c.plugin.Status()
-}
-
 // toCNIPortMappings converts CRI port mappings to CNI.
 func toCNIPortMappings(criPortMappings []*runtime.PortMapping) []ocicni.PortMapping {
```
Comment (YaoZengzeng) [inline cri/v1alpha1/cri_network.go]: Maybe move this function to `cri_utils.go` and rm this file.

File: cri/v1alpha2/cri_network.go [hunk 5]
```diff
@@ -1,130 +1,13 @@
...
-	return ip, nil
-}
-
-// Status returns error if the network plugin is in error state.
-func (c *CniManager) Status() error {
-	return c.plugin.Status()
-}
-
 // toCNIPortMappings converts CRI port mappings to CNI.
 func toCNIPortMappings(criPortMappings []*runtime.PortMapping) []ocicni.PortMapping {
```
Comment (YaoZengzeng) [inline cri/v1alpha2/cri_network.go]: The same as above.

File: cri/v1alpha1/cri_network.go (hunk 4 above)
Comment (Starnop) [reply cri/v1alpha1/cri_network.go]: Well, maybe we'll consider creating separate files in the future when we have more utility functions.

File: cri/v1alpha2/cri_network.go (hunk 5 above)
Comment (Starnop) [reply cri/v1alpha2/cri_network.go]: Done.

**Comment by allencloud** [comment]
PTAL @YaoZengzeng

**Comment by YaoZengzeng** [comment]
LGTM

**Pull request merged**